hermeto/core/resolver.py

```diff
@@ -64,14 +64,31 @@ def _resolve_packages(request: Request) -> RequestOutput:
     """Run all requested package managers, return their combined output."""
     _supported_package_managers = _package_managers
     requested_types = set(pkg.type for pkg in request.packages)
-    if "dev-package-managers" in request.flags:
-        _supported_package_managers = _package_managers | _dev_package_managers
+
+    # Convert x-<pkg> -> <pkg> and track which ones are experimental
+    # Use str instead of PackageManagerType since we're working with runtime values
+    normalized_types: set[str] = set()
+    experimental_types: set[str] = set()
+    for t in requested_types:
+        if not refers_to_experimental_pm(t):
+            normalized_types.add(t)
+        else:
+            # x-foo -> foo
+            normalized_type = t[2:]  # Remove "x-" prefix
+            normalized_types.add(normalized_type)
+            experimental_types.add(normalized_type)
+
+    if experimental_types:
+        requested_types = normalized_types  # type: ignore[assignment]
```

Comment:
Could you please add a comment as of why type check has to be ignored? I believe `mypy` insists that left-hand side is too narrow because of name re-use but this is a scratch variable anyway? The same stands for L85.